veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,214 @@
+# VEP #29: NIC Hotplug / Hotunplug For Namespace Admins Following VMIM Hardening
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+Currently, the NIC hotplug / hotunplug feature depends on a user-initiated migration following each NIC hotplug / hotunplug
+operation.
+The migration is required in order to attach / detach secondary interfaces to / from the target virt-launcher pod.
+
+> [!NOTE]
+> KubeVirt also supports in-place hotplug / hotunplug operations (without performing a migration)
+> by integrating with the [dynamic-networks-controller](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller).
+>
+> This method does not support hot-plugging SR-IOV NICs, as SR-IOV hotplug operation requires a migration in order for the
+> [SR-IOV device plugin](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin) to be invoked when the
+> target pod is created.
+
+PR [kubevirt/kubevirt#13497](https://github.com/kubevirt/kubevirt/pull/13497) removed Namespace administrators'
+permissions to create VirtualMachineInstanceMigration objects - thus breaking the NIC hotplug / hotunplug feature
+for them in v1.5.
+
+## Motivation
+
+In previous KubeVirt versions, a namespace admin could hotplug and unplug NICs.
+We want to ensure that this feature will continue working in v1.5 and future versions.
+
+## Goals
+
+Enable namespace admins to perform NIC hotplug / unplug operations independently - without involving the cluster-admin 
+for each hotplug / hotunplug operation.
+
+## Non Goals
+
+## Definition of Users
+
+- Namespace administrators
+
+## User Stories
+
+- As a namespace admin I would like to be able to hotplug / unplug a secondary interface without involving the cluster
+  admin.
```

Comment:
Please list the different scenarios:
- Hotplug a new secondary network interface that uses bridge binding.
- Hotunplug a secondary network interface that uses bridge binding.
- Hotplug a new secondary network interface of type SR-IOV.
- Disallow hotplug of a primary network interface. (or mark that it requires reset?)
- Disallow hotunplug of a primary network interface.  (or mark that it requires reset?)
- Allow using a staging strategy so any hotplug/hotunplug will take effect only after restarting a VM.

Reply:
Added all of them but the following two:
- Disallow hotplug of a primary network interface. (or mark that it requires reset?)
- Disallow hotunplug of a primary network interface. (or mark that it requires reset?)

Currently, the NIC hotplug feature only supports secondary networks, but doesn't block the addition of primary interfaces.
Lets discuss it first, prior to adding it here.
We could also add SR-IOV NIC hotunplug, which is currently unsupported.

Reply:
Hotplugging  / unplugging a primary NIC is also not covered by e2e tests.

Reply:
There is still a need to define what should happen when one attempts to perform actions on the primary.
You could define that they need to be rejected or to mark it for restart. Leaving it undefined is not nice.